### PR TITLE
WIP: Docker backend

### PR DESCRIPTION
**Pull request by mikicz**

### CODE REVIEW

**Comment by hroncok** [comment]
I can do a review tmrw, if you are interested.

**Comment by mikicz** [comment]
I'm not quite done yet, I'll message you if I'm gonna be done with this today, but yes, a review would be nice. Thanks!

**Comment by mikicz** [comment]
@hroncok Please take a look at this if you find the time, I'm not quite done yet, but any thoughs will be appreciated

**Comment by hroncok** [comment]
On it today. sorry for the delay. Still seem unfinished anyway.

File: arca/backend/docker.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+class DockerBackend(BaseBackend):
+
+    python_version = LazySettingProperty(key="python_version", default=None)
+    keep_container_running = LazySettingProperty(key="keep_container_running", default=False)
+    apk_dependencies = LazySettingProperty(key="apk_dependencies", default=None)
+    disable_pull = LazySettingProperty(key="disable_pull", default=False)  # so the build can be tested
+
+    def __init__(self, **kwargs):
+        super(DockerBackend, self).__init__(**kwargs)
```
Comment (hroncok) [inline arca/backend/docker.py]: Are you aiming for Python 2 compatibility? If not, maybe just use `super()`.

File: arca/backend/docker.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        try:
+            res = container.exec_run(["python", f"/srv/scripts/{script_name}"], tty=True)
+
+            return Result(json.loads(res))
+        except Exception as e:
+            logger.exception(e)
+            return Result({"success": False, "error": str(e)})
+        finally:
+            if not self.keep_container_running:
+                container.kill("9")
```
Comment (hroncok) [inline arca/backend/docker.py]: Maybe get the number from some constant? `signal.SIGKILL` seems to be more descriptive than hardcoded 9.

File: tests/backend/test_backends.py [hunk 3]
```diff
@@ -26,17 +26,17 @@ def return_str_function():
...
+        (None, "requirements/requirements.txt"),
         (None, "test_package"),
-        ("requirements/requirements.txt", "test_package"),
-    ]
+    ))
 )
-def test_venv_backend(requirements_location, file_location):
+def test_backends(backend, requirements_location, file_location):
+    if os.environ.get("TRAVIS", False) and backend == VenvBackend:
+        raise pytest.skip("Venv Backend doesn't work on Travis")
```
Comment (hroncok) [inline tests/backend/test_backends.py]: Should you raise this or call this? The call raises an exception.

File: arca/backend/docker.py (hunk 1 above)
Comment (mikicz) [reply arca/backend/docker.py]: Oh, nope, not at all, replaced, PyCharm autocompletion did that...

File: arca/backend/docker.py (hunk 2 above)
Comment (mikicz) [reply arca/backend/docker.py]: Yep, replaced

File: tests/backend/test_backends.py (hunk 3 above)
Comment (mikicz) [reply tests/backend/test_backends.py]: Yep, indeed, the call raises an exception so therefore the raise keyword is useless.

**Pull request merged**